### PR TITLE
Allow single input for merging layers Add, Average, Concatenate, Maximum, Minimum, Multiply

### PR DESCRIPTION
**Pull request by foxik**
The merging layers (notably `Add`, `Average`, `Concatenate`, `Maximum`, `Minimum`, `Multiply`) require at least two inputs.

This is already documented in the lowercase versions of the layers (like `add`), but is missing for `Layer` instances (like `Add`).

This PR adds this information also to the documentation of the `Layer` instances.

Also, is there any reason why the single input is not supported? For all the six mentioned layers, a single input would work fine -- both mathematically and in the current implementation,the only required change would be to remove the error in the
https://github.com/keras-team/keras/blob/980a6be629610ee58c1eae5a65a4724ce650597b/keras/layers/merging/base_merge.py#L86-L90.

### CODE REVIEW

**Comment by fchollet** [comment]
(We would also need corresponding unit tests in this case)

**Comment by fchollet** [comment]
Excellent, thank you!

**Comment by foxik** [comment]
I have changed the commit to allow passing a single element to the mentioned 6 layers.

- I removed the "at least 2" from the documentation.
- I removed two tests checking that `tf.keras.layers.add()[i1])` fails.
- I added a test checking that single-element list works for `Add`, `add`, `Average`, `average`, `Concatenate`, `contatenate`, `Maximum`, `maximum`, `Minimum`, `minimum`, `Multiply`, `multiply`
- The input is still required to be a list; a test requiring the above layers to fail is included

File: keras/layers/merging/merging_test.py [hunk 1]
```diff
@@ -271,6 +272,35 @@ def test_merging_with_scalar_input(self, layer):
...
+    # Instantiate the Layer subclasses
+    if tf_inspect.isclass(layer) and issubclass(layer, keras.layers.Layer):
+      layer = layer()
+
+    # Processing a single element list should behave as identity.
+    i1 = keras.layers.Input(shape=(4, 5))
+    o = layer([i1])
+    self.assertListEqual(o.shape.as_list(), [None, 4, 5])
+    model = keras.models.Model(i1, o)
+    model.run_eagerly = test_utils.should_run_eagerly()
```
[the remark this answers is not in the document]
Comment (foxik) [reply keras/layers/merging/merging_test.py:293]: I will gladly remove it; but it is used all over the file, for example in
- https://github.com/keras-team/keras/blob/b67f90fa06176f6e0da23681f53dad9df7e88949/keras/layers/merging/merging_test.py#L39
- https://github.com/keras-team/keras/blob/b67f90fa06176f6e0da23681f53dad9df7e88949/keras/layers/merging/merging_test.py#L71
- https://github.com/keras-team/keras/blob/b67f90fa06176f6e0da23681f53dad9df7e88949/keras/layers/merging/merging_test.py#L104

to name a few. I expected that the decorator https://github.com/keras-team/keras/blob/b67f90fa06176f6e0da23681f53dad9df7e88949/keras/layers/merging/merging_test.py#L27 is making the `should_run_eagerly` return different results; also, I just verified that in the testing logs I see both tests with `test_utils.should_run_eagerly()` returning `True` (e.g., `test_single_element_test_layer_classkeraslayersmergingaverageAverage_v2_eager`) and `False` (e.g., `test_single_element_test_layer_classkeraslayersmergingaverageAverage_v2_function`).

If it is in really an error, should I also remove the other occurrences?

File: keras/layers/merging/merging_test.py [hunk 2]
```diff
@@ -271,6 +272,35 @@ def test_merging_with_scalar_input(self, layer):
...
+
+    x1 = np.random.random((2, 4, 5))
+    out = model.predict(x1)
+    self.assertEqual(out.shape, (2, 4, 5))
+    self.assertAllClose(out, x1)
+
+    # A single element must be passed as a list, not by itself.
+    with self.assertRaises(ValueError):
```
[the remark this answers is not in the document]
Comment (foxik) [reply keras/layers/merging/merging_test.py]: Good idea, done. I took the error message from https://github.com/keras-team/keras/blob/b67f90fa06176f6e0da23681f53dad9df7e88949/keras/layers/merging/merging_test.py#L321-L322

**Comment by foxik** [comment]
Thanks for the review :-)

File: keras/layers/merging/merging_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fchollet) [reply keras/layers/merging/merging_test.py:293]: SG, then we're good.

**Review by fchollet — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**